x/incentives/client/cli/query.go

```diff
@@ -384,31 +386,77 @@ $ %s query incentives rewards-estimation
 				return err
 			}
 
-			lockIdStrs := strings.Split(lockIdsCombined, ",")
-			lockIds := []uint64{}
-			for _, lockIdStr := range lockIdStrs {
-				lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+			var res *lockuptypes.AccountLockedLongerDurationResponse
+			if owner != "" {
+				queryClientLockup := lockuptypes.NewQueryClient(clientCtx)
+
+				res, err = queryClientLockup.AccountLockedLongerDuration(cmd.Context(), &lockuptypes.AccountLockedLongerDurationRequest{Owner: owner, Duration: time.Millisecond})
 				if err != nil {
 					return err
 				}
-				lockIds = append(lockIds, lockId)
+			} else {
+				owner = ""
+			}
+
+			ownerLocks := []uint64{}
+
+			for _, lockId := range res.Locks {
+				ownerLocks = append(ownerLocks, lockId.ID)
+			}
+
+			lockIdStrs := strings.Split(lockIdsCombined, ",")
+			lockIds := []uint64{}
+			// if user doesn't provide at least one of the lock ids or owner, we don't have enough information to proceed.
+			if lockIdsCombined == "" && owner == "" {
+				return fmt.Errorf("if owner flag is not set, lock IDs must be provided")
+
+				// if user provides lockIDs, use these lockIDs in our rewards estimation
+			} else if owner == "" {
+				for _, lockIdStr := range lockIdStrs {
+					lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+					if err != nil {
+						return err
+					}
+					lockIds = append(lockIds, lockId)
+				}
+
+				// if no lockIDs are provided but an owner is provided, we query the rewards for all of the locks the owner has
+			} else if lockIdsCombined == "" {
+				lockIds = append(lockIds, ownerLocks...)
+			}
+
+			// if lockIDs are provided and an owner is provided, only query the lockIDs that are provided
+			// if a lockID was provided and it doesn't belong to the owner, return an error
+			if len(lockIds) != 0 && owner != "" {
+				for _, lockId := range lockIds {
```

Comment:
How can this work when no owner is provided?

If no owner is provided, then `ownerLocks` are empty and we only use locks from `lockIdStrs`. However, since `ownerLocks` are empty, `validInputLockId` would be false for any lock supplied via a flag string